libraries/encrypted-db/src/main/kotlin/io/element/encrypteddb/passphrase/PassphraseProvider.kt

```diff
@@ -19,7 +19,7 @@ package io.element.encrypteddb.passphrase
 /**
  * An abstraction to implement secure providers for SQLCipher passphrases.
  */
-interface PassphraseProvider {
+fun interface PassphraseProvider {
```

Comment:
I do not think we want to have a `fun interface` here and for the other 2 places you applied the same change. `fun interface` are more useful when we have unnamed object that implement them. This is not the case here, and the interfaces may be extended at some point.